arviz/data/inference_data.py

```diff
@@ -589,6 +586,108 @@ def to_dataframe(
             (dfs,) = dfs.values()
         return dfs
 
+    def to_zarr(self, store=None):
+        """
+        Convert InferenceData to a :class:`zarr.hierarchy.group` using the same groups names
+        as in InferenceData.
+
+        Data groups ("observed_data", "constant_data", "predictions_constant_data") are
+        skipped.
+
+        Raises TypeError if no valid store is found.
+
+        Parameters
+        ----------
+        store: zarr.storage i.e MutableMapping or str, optional
+            Zarr storage class or path to desired DirectoryStore.
+
+        overwrite: bool, optional,
+            If True, delete any pre-existing data in store at path before creating the groups.
+
+        Returns
+        -------
+        zarr.hierarchy.group
+            A zarr hierarchy group containing the InferenceData.
+
+        References
+        ----------
+        https://zarr.readthedocs.io/
+        """
+
+        import zarr
```

Comment:
Is zarr always installed? 

If it is optional import, then I think we can raise Import Error and add some text to tell user to install zarr. I think we do similar thing with bokeh.

Reply:
I put zarr into the optional imports, will take a look into the bokeh functions to create something similar.